api/src/services/import-services/sampling-periods/import-sample-periods-service.ts

```diff
@@ -0,0 +1,144 @@
+import { TechniqueService } from '../../technique-service';
+import { getMethodTechniqueCellValidator, getSampleSiteCellValidator } from './utils/sample-periods-header-configs';
+
+const defaultLog = getLogger('services/import/import-sample-periods-service');
+
+// Sample Period CSV static headers
+export type SamplePeriodCSVStaticHeader =
+  | 'SAMPLE_SITE'
+  | 'TECHNIQUE_NAME'
+  | 'START_DATE'
+  | 'START_TIME'
+  | 'END_DATE'
+  | 'END_TIME';
+
+/**
+ * ImportSamplePeriodsService - A service for importing Sample Periods into SIMS.
+ *
+ * @class ImportSamplePeriodsService
+ * @extends DBService
+ */
+export class ImportSamplePeriodsService extends DBService {
+  worksheet: WorkSheet;
+  surveyId: number;
+
+  utils: CSVConfigUtils<SamplePeriodCSVStaticHeader>;
+
+  /**
+   * Construct an instance of ImportSamplePeriodsService.
+   *
+   * @param {IDBConnection} connection - DB connection
+   * @param {string} surveyId
+   */
+  constructor(connection: IDBConnection, worksheet: WorkSheet, surveyId: number) {
+    super(connection);
+
+    const initialConfig: CSVConfig<SamplePeriodCSVStaticHeader> = {
+      staticHeadersConfig: {
+        SAMPLE_SITE: { aliases: ['SAMPLE SITE', 'SAMPLING_SITE', 'SAMPLING SITE', 'SITE', 'LOCATION', 'STATION'] },
+        TECHNIQUE_NAME: { aliases: ['TECHNIQUE NAME', 'METHOD_TECHNIQUE', 'METHOD TECHNIQUE', 'TECHNIQUE', 'METHOD'] },
+        START_DATE: { aliases: ['START DATE'] },
+        START_TIME: { aliases: ['START TIME'], optional: true },
+        END_DATE: { aliases: ['END DATE'] },
+        END_TIME: { aliases: ['END TIME'], optional: true }
+      },
+      ignoreDynamicHeaders: false
+    };
+
+    this.worksheet = worksheet;
+    this.surveyId = surveyId;
+
+    this.utils = new CSVConfigUtils(this.worksheet, initialConfig);
+  }
+
+  /**
+   * Import a `Sample Periods` CSV worksheet into SIMS.
+   *
+   * @async
+   * @throws {ApiGeneralError} - If unable to fully insert records into Critterbase
+   * @returns {*} {Promise<CSVError[]>} List of CSV errors encountered during import
+   */
+  async importCSVWorksheet(): Promise<CSVError[]> {
+    const config = await this.getCSVConfig();
+
+    const { errors, rows } = validateCSVWorksheet(this.worksheet, config);
+
+    if (errors.length) {
+      return errors;
+    }
+
+    const samplePeriodService = new SamplePeriodService(this.connection);
+
+    // Convert the rows to sample periods
+    const samplePeriods: InsertSamplePeriodObject[] = rows.map((row) => {
+      return {
+        survey_sample_site_id: row.SAMPLE_SITE,
+        method_technique_id: row.TECHNIQUE_NAME,
+        start_date: row.START_DATE,
+        start_time: row.START_TIME,
+        end_date: row.END_DATE,
+        end_time: row.END_TIME
+      };
+    });
+
+    await samplePeriodService.insertSamplePeriods(this.surveyId, samplePeriods);
+
+    defaultLog.debug({ label: 'import sample periods', samplePeriods });
```

Comment:
Re logging in general: We don't have a strict pattern established per se, but I think it might be worth including the name of the function in the log:

```
defaultLog.debug({ label: 'importCSVWorksheet', message: 'import sample periods', samplePeriods });
```

That combined with the label that gets set at the file level when you call getLogger helps narrow down where the log message came from more precisely.

Reply:
I was going to make the label and message fields required as part of that previous logging ticket, but there are some logs in the app that dont have them, and I didnt want to make a giant PR

Reply:
Updated.